docs/Custom-brimcap-load-Config.md

```diff
@@ -0,0 +1,433 @@
+remove this part of the pipeline. If so, you'll likely want to explore the
+additional data and create shapers to apply proper data types to them, since
+this will be a prerequisite for doing certain Zed queries with the data (e.g.
+a successful CIDR match requires IP addresses to be stored as `ip` type, not
+the string type in which they'd appear in unshaped JSON).
+
+3. The `put . = shape(alert)` applies the shape of the `alert` type to each
+input record. With what's shown here, additional fields that appear beyond
+those specified in the shaper (e.g. as the result of new Suricata features or
+your own customizations) will still be let through this pipeline and stored in
+the Zed Lake. If this is undesirable, add `| put . = crop(alert)` downstream of
+the first `put`, which will trim these additional fields.
+
+4. The `rename ts=timestamp` changes the name of Suricata's `timestamp` field
+to match the `ts` one used by Zeek, which allows the data from both sources to
+be more seamlessly presented together in Brim.
+
+# Custom NetFlow Analyzer
+
+## Background
+
+Building on what we just learned about Brimcap analyzers, we can extend the
+concepts to pcap processing tools other than Zeek or Suricata. In this case
+we'll use NetFlow records generated using the open source
+[nfdump](https://github.com/phaag/nfdump) toolset.
+
+* **Note**: While the example shows how shaped NetFlow records can be loaded
+via Brimcap, the **Packets** button in Brim is not currently wired to extract
+flows from a pcap via the 5-tuple and time details in NetFlow records. If this
+is functionality you're interested in pursuing, please
+[open an issue](https://github.com/brimdata/brimcap/wiki/Troubleshooting#opening-an-issue)
+or come talk to us on [public Slack](https://www.brimsecurity.com/join-slack/).
+
+## Base nfdump Installation
+
+To build and install the nfdump toolset with the commands we'll need to
+generate NetFlow, we execute the following on our Linux host:
+
+```
+sudo apt install automake libtool pkg-config flex bison libbz2-dev
+git clone https://github.com/phaag/nfdump.git && cd nfdump
+./autogen.sh
+./configure --enable-nfpcapd
+sudo make install
+export LD_LIBRARY_PATH="/usr/local/lib"
+```
+
+## Example Configuration
+
+As we did with Zeek and Suricata, we create a wrapper script to act as our
+Brimcap analyzer. It works in two phases, first creating binary NetFlow records
+and then converting them to CSV. `nfpcapd` only accepts a true pcap file input
+(not a device like `/dev/stdin`), so we first store the incoming pcap in a
+temporary file. We also use GNU `head` to trim out some trailing summary lines
+that `nfdump` appends by default which would trip up the Zed CSV reader.
+
+```
+$ cat nfdump-wrapper.sh 
+TMPFILE=$(mktemp)
+cat - > "$TMPFILE"
+nfpcapd -r "$TMPFILE" -l .
+rm "$TMPFILE"
+for file in nfcapd.*
+do
+  nfdump -r $file -o csv | head -n -3 | zq -i csv -f ndjson - > ${file}.ndjson
```

Comment:
I don't actually mention the `zq -i csv -f ndjson` step in the text because I'm hoping I don't have to publish with it. Here's the reasons why this wrapper is hacky at the moment:

1.  `nfdump` has a JSON output option, but it's an _array_ of JSON objects, not NDJSON. When I tried to use our recently-added `-i json` mode to read it, I easily exceeded our current JSON reader limits. As @mccanne mentioned at sync, perhaps we could enhance the JSON input mode to recognize when it's reading an array and be more stream-like in its read behavior? If that were the case, I'd switch from CSV to JSON here.  
     
2.  As an alternative to JSON, CSV mostly works, but that hits a different snag. When we get to the phase where Brimcap needs to apply the shaper, it fails with CSV because I don't see a way to do the equivalent of `-i csv` when running `brimcap load`.

Because of those two problems, here I end up turning the CSV back into NDJSON in the wrapper script. Yes, I know I could have applied the shaper right here in this `zq` portion and turn it directly into ZNG, but that would defeat the purpose of showing how Brimcap uses shapers.

Reply:
Since it looks like neither #80 nor brimdata/zed#2517 are going to make it into MVP0, for now I've adjusted the wrapper script to turn the CSV into minimally-typed ZNG and let that be the input to our shaping step. I included a "Note:" block explaining why this was done.